### PR TITLE
separate compile-test-handler from test-handler and add env

### PR DESCRIPTION
**Pull request by k13gomez**
- separates compile-test-handler from compile-test, this separates this functionality similar to how it was prior to compile-test being changed into it's current form and helps with advanced profiling techniques using alter-var-root to instrument compiled nodes.
- add env support to compile-test which was noted as something to fix in the future and was fairly simple to add

### CODE REVIEW

**Review by WilliamParker — COMMENTED** [review]
I don't see any functional problems here, although we should probably have a test.  Although for that matter, I looked for tests where the env is used for other condition types and didn't find any.  Unless I'm missing them somewhere (there are a lot of tests, I'd like to have another look later) perhaps we should have an issue to add those tests.

File: src/main/clojure/clara/rules/compiler.clj
Comment (k13gomez) [reply src/main/clojure/clara/rules/compiler.clj:435]: it was previously wrapped in a `let` binding call, the problem steps from the fact that there is schema validation happening with these expressions that require the output of the `compile-*` functions to be a s-expr, returning a map literal breaks that contract, and this seemed like an easy compromise. Functionally it works the same way as it did before minus the let-binding layer.

**Comment by k13gomez** [comment]
This is a work in-progress, I am adding some tests and ensuring they pass. One thing I have noticed as well is that there are other areas where `env` is actually not passed through and an empty map is passed, I am considering if I should add more extensive tests for uses of `env` and then fix the remaining places where `env` is passed as `{}`

File: CHANGELOG.md
Comment (WilliamParker) [inline CHANGELOG.md:3]: Minor: Perhaps change to 0.23.0-SNAPSHOT and then back to 0.23.0 when the release is cut so that anyone reading this isn't confused into thinking 0.23.0 has been released when it hasn't.  This probably doesn't matter if you're planning to release 0.23.0 upon merging this though.

File: src/main/clojure/clara/rules/compiler.clj
Comment (WilliamParker) [reply src/main/clojure/clara/rules/compiler.clj:435]: OK, makes sense.

File: src/main/clojure/clara/macros.clj
Comment (WilliamParker) [inline src/main/clojure/clara/macros.clj:135]: I think this would have fixed the ClojureScript compiler error I see in the build that failed? https://github.com/cerner/clara-rules/actions/runs/5136943790/jobs/9244352594

File: src/test/clojure/clara/test_rules.clj
Comment (WilliamParker) [inline src/test/clojure/clara/test_rules.clj:1343]: I think we can just have a RHS here that doesn't really do anything.   For example an operation like
```(inc 1)``` or anything similar.  In order to resemble real rules it is probably better to have something to execute though rather than say ```(do nil)``` although that would work too.

 This adds some clutter to the logs of the test build output.

From https://github.com/cerner/clara-rules/actions/runs/5152272254/jobs/9278212444 :
````
lein test clara.test-rules
42
42
````

**Review by WilliamParker — APPROVED** [review]
Not sure if this still a draft @k13gomez - I left some minor comments but nothing very major.

**Review by WilliamParker — APPROVED** [review]

File: src/main/clojure/clara/macros.clj
Comment (k13gomez) [reply src/main/clojure/clara/macros.clj:135]: Yes, I had missed this in my initial pass.

File: src/test/clojure/clara/test_rules.clj
Comment (k13gomez) [reply src/test/clojure/clara/test_rules.clj:1343]: Makes sense, will make this change.

File: CHANGELOG.md
Comment (k13gomez) [reply CHANGELOG.md:3]: Will do since I might wait until I also merge #487 to do a release.

**Comment by k13gomez** [comment]
@WilliamParker I also noticed there are actually other places where `env` is not passed through, I am planning to open a separate PR for that, but not in a rush to do it, the places where env isn't passed through are these:
- https://github.com/cerner/clara-rules/blob/main/src/main/clojure/clara/rules/engine.cljc#L673
- https://github.com/cerner/clara-rules/blob/main/src/main/clojure/clara/rules/engine.cljc#L1382

An empty map is passed in both of these cases, I'd like your view on whether this is worth fixing or if it was done in this way by design.

**Pull request merged**
